### PR TITLE
fix: align ISODate with shell COMPASS-4655

### PR DESCRIPTION
**Pull request by mcasimir**
Currently we treat `IsoDate` as `Date`.

`IsoDate` works differently and always construct dates ignoring the current time zone if one is not specified.

The code in the PR is copy-pasted from mongosh to make sure we address this type in the same way.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -2,8 +2,8 @@
   "compilerOptions": {
     /* Basic Options */
     // "incremental": true,                   /* Enable incremental compilation */
-    "target": "es5",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019' or 'ESNEXT'. */
-    "module": "es2015",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */
+    "target": "ES2018",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019' or 'ESNEXT'. */
```
Comment (mcasimir) [inline tsconfig.json:5]: should be safe at this point? helps with regex groups

File: package-lock.json [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
   "name": "ejson-shell-parser",
-  "version": "2.0.0",
+  "version": "2.0.1",
   "lockfileVersion": 2,
   "requires": true,
   "packages": {
     "": {
       "name": "ejson-shell-parser",
-      "version": "2.0.0",
+      "version": "2.0.1",
```
Comment (addaleax) [inline package-lock.json:9]: We still do releases through `npm version patch && git push origin main --tags`?

**Review by addaleax — APPROVED** [review]

File: package-lock.json (hunk 2 above)
Comment (mcasimir) [reply package-lock.json:9]: Not sure. I don't think I've released this in years, how does it relate to this line?

**Comment by mcasimir** [comment]
Also yeah, CI fails https://github.com/mongodb-js/ejson-shell-parser/actions/runs/9659500397/job/26643024959

File: package-lock.json (hunk 2 above)
Comment (addaleax) [reply package-lock.json:9]: It just means we usually do the version bumps outside of PRs 🙂 Definitely not a big deal (hence the approval), you can do it whichever way you want

**Comment by addaleax** [comment]
> Also yeah, CI fails https://github.com/mongodb-js/ejson-shell-parser/actions/runs/9659500397/job/26643024959

Yeah, this package uses yarn, I think the package lock/yarn lock changes here are being rejected by it 🥴

**Pull request merged**